### DISCUSSION TITLE
why sprite outside canvas sprite can screenshot

### DISCUSSION THREAD

**Discussion by a6965921** [pixijs/pixijs#7196] [Q&A]
why sprite outside canvas sprite can screenshot

     var jietu = app.renderer.extract.base64(app.stage);
     sprite outside canvas   it has white background

**Comment by xzilja** [comment]
I believe this could be a problem related to culling. Pixi always renders and keeps elements in scene graph even if they are outside of canvas. It is your job to not render them by setting [visible](https://pixijs.download/dev/docs/PIXI.Container.html#visible) and / or [renderable](https://pixijs.download/dev/docs/PIXI.Container.html#renderable) attributes. Often this is called "culling" and there are some packages that make this easier https://github.com/davidfig/pixi-cull
